esphome/components/m5stack4relay/switch/__init__.py

```diff
@@ -0,0 +1,61 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c, switch
+from esphome.const import CONF_CHANNEL, CONF_INTERLOCK  # , CONF_ID,
+
+from .. import m5stack4relay_ns, M5Stack4Relay, CONF_M5STACK4RELAY_ID
+
+DEPENDENCIES = ["m5stack4relay"]
+
+M5StackSwitch = m5stack4relay_ns.class_(
+    "M5Stack4RelaySwitch", cg.Component, i2c.I2CDevice, switch.Switch
+)
+
+CONF_INTERLOCK_WAIT_TIME = "interlock_wait_time"
+
+CONF_Relay_1 = 1
+CONF_Relay_2 = 2
+CONF_Relay_3 = 3
+CONF_Relay_4 = 4
+
+RelayBit_ = m5stack4relay_ns.enum("RelayBit", is_class=True)
+
+SWITCH_MAP = {
+    CONF_Relay_1: RelayBit_.RELAY1,
+    CONF_Relay_2: RelayBit_.RELAY2,
+    CONF_Relay_3: RelayBit_.RELAY3,
+    CONF_Relay_4: RelayBit_.RELAY4,
+}
+
+
+CONFIG_SCHEMA = (
+    switch.switch_schema(M5StackSwitch)
+    .extend(
+        {
+            cv.GenerateID(): cv.declare_id(M5StackSwitch),
+            cv.GenerateID(CONF_M5STACK4RELAY_ID): cv.use_id(M5Stack4Relay),
+            cv.Required(CONF_CHANNEL): cv.enum(SWITCH_MAP),
```

Comment:
Since the channel is not very complicated, lets just keep it as an integer
```suggestion
            cv.Required(CONF_CHANNEL): cv.int_range(min=1, max=4),
```